ui/component/viewers/videoViewer/index.js

```diff
@@ -40,6 +40,16 @@ const select = (state, props) => {
   const uri = props.uri;
 
   const claim = selectClaimForUri(state, uri);
+  const { signing_channel } = claim || {};
+  let channelTitle = null;
+  if (signing_channel) {
+    const { value, name } = signing_channel;
+    if (value && value.title) {
+      channelTitle = value.title;
+    } else {
+      channelTitle = name;
+    }
+  }
```

Comment:
Not a big deal, but just some info for the future consideration:

- The state-to-props mapping happens frequently (every time Redux updates), so this is run a lot.
  - That is why some of the selectors with heavy transformation are cached (`createCachedSelector`, `createSelector`)
  - For this case, it's just accessing properties, so doesn't really matter whether it is cached or not, and not a big deal.  But just highlighting things to consider when filling up `select`.
- Also for this case, the entire `claim` is already passed to the jsx, so we can actually determine the `channelTitle` there instead.  Benefits:
  - The jsx will only run when there are prop and state changes, whereas this side of the code runs on every Redux update.
  - One less prop for React to diff for changes.
- Side note: passing entire objects like `claim` as a prop is usually not efficient as object references can get invalidated (even if the content is identical), so it will be regarded as "changed" and causes a render.  But on the other hand, passing the entire `claim` is cleaner and easier to code if the component needs to access many properties.

Summary:  this mapping stage was one of the sources of performance issues that we cleaned up last year (not 100%), so it's something to keep in mind for future code.

Reply:
Aight, I moved it to the view.jsx like this and it seems to work fine. Thanks!

`const channelTitle = claim && claim.signing_channel && claim.signing_channel.value && claim.signing_channel.value.title;`